packages/media-utils/src/utils/flatten-form-data.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Recursively flatten data passed to form data, to allow using multi-level objects.
+ *
+ * @param {FormData}      formData Form data object.
+ * @param {string}        key      Key to amend to form data object
+ * @param {string|Object} data     Data to be amended to form data.
+ */
+export function flattenFormData(
+	formData: FormData,
+	key: string,
+	data: string | undefined | Record< string, string >
+) {
+	if (
+		typeof data === 'object' &&
+		Object.getPrototypeOf( data ) === Object.prototype
+	) {
+		for ( const name in data ) {
+			if ( Object.prototype.hasOwnProperty.call( data, name ) ) {
```

Comment:
`for ... in` will iterate over all enumerable string properties of an object. So why is it necessary to have the `hasOwnProperty` check?

Reply:
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#iterating_own_properties

(I'll update to use `hasOwn()`, which is nowadays recommended over `Object.prototype.hasOwnProperty.call`)